my-webxr-app/tests/data/DataLayer.test.tsx

```diff
@@ -195,6 +195,54 @@ describe('Validate storeCSV() operation', () => {
     await Dexie.delete(dbName);
   });
 
+  test('given a empty batch, expect exception', async () => {
+    const resolve = await dataLayer.storeCSV([]);
+    expect(resolve).toEqual(false);
+
+    const isEmpty = await repository.isTableEmpty(TableName.RAW);
+
+    expect(isEmpty).toBeTruthy();
+  });
+
+  test('given a batch with headers of type number, headers should be cast to type string', async () => {
+    const batchItems: BatchedDataStream = [[1, 2],
+      [-100, -101]];
+
+    const resolve = await dataLayer.storeCSV(batchItems);
+    expect(resolve).toEqual(true);
+
+    const column1 = await repository.getColumn(String(1), TableName.RAW);
+    const column2 = await repository.getColumn(String(2), TableName.RAW);
+
+    expect(column1.values).toEqual([-100]);
+    expect(column2.values).toEqual([-101]);
+  });
+
+  test('given a batch of only one header', async () => {
+    const batchItems: BatchedDataStream = [['column1']];
+
+    const resolve = await dataLayer.storeCSV(batchItems);
+    expect(resolve).toEqual(true);
+
+    const column1 = await repository.getColumn('column1', TableName.RAW);
+    expect(column1.values).toHaveLength(0);
+  });
+
+  test('should store a batch with only header row correctly', async () => {
+    const batchItems: BatchedDataStream = [
+      ['column1', 'column2'],
+    ];
+
+    const resolve = await dataLayer.storeCSV(batchItems);
+    expect(resolve).toEqual(true);
+
+    const column1 = await repository.getColumn('column1', TableName.RAW);
+    const column2 = await repository.getColumn('column2', TableName.RAW);
+
+    expect(column1.values).toHaveLength(0);
+    expect(column2.values).toHaveLength(0);
+  });
+
```

Comment:
Great tests!

Do we need to test if batch data stream entires is null?

Reply:
I don't think we need to since in Typescript. Null and Undefined aren't a value of an object but are rather a type. Since we already restrict the type of argument to BatchedDataStream (which can only be wither Array[string] or Array[number], we're already sure that the variable must be of type BatchedDataStream (not any other type like Null or Undefined) ?

Reply:
TypeScript is weird ngl. I love the idea, but still hate it. As Trang's comment, we don't need to check it, given TypeScript does it job